tests/qa_automation/kernel_xfstests.pm

```diff
@@ -0,0 +1,63 @@
+# Copyright (C) 2017 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary:  xfstests testsuite
+# Use the latest xfstests testsuite from upstream to make file system test
+# Maintainer: Yong Sun <yosun@suse.com>
+
+use base "qa_run";
+use strict;
+use testapi;
+
+sub run() {
+    my $self = shift;
+    $self->system_login();
+    $self->prepare_repos();
+    # create test partition
+    assert_script_run("parted -l -m 2>&1");
+    my $cmd           = "parted -l -m 2>&1| awk -F \':\' \'{if(\$5 == \"xfs\") print \$1}\'";
+    my $partition_num = $self->qa_script_output($cmd, 10);
+    my $partition     = "/dev/vda" . $partition_num;
+    assert_script_run("fdisk -l " . $partition, 10);
```

Comment:
Acturally you don't need add timeout here for assert_script_run if you can expect the command finish executing within 90s.